src/Eloquent/Connection.php

```diff
@@ -134,6 +134,15 @@ public function zohoUpsert(string $toTable, array $data, array|string $key): int
             $key = [$key];
         }
 
+        foreach ($data as $rowIndex => $row) {
+            foreach ($row as $field => $value) {
+                if (is_object($value) && method_exists($value, '__toString')) {
```

Comment:
This can be simplified to something like:

```php
$row[$field] = $value instanceof \Stringable ? (string) $value : $value;
```
The `Stringable` interface is implicitly available on any object with a `__toString()` method.